### PR TITLE
Highlight toggle and command

### PR DESCRIPTION
**Pull request by Nick-de-Bruin**
Adds a highlight toggle in the settings, and adds a command to toggle it on or off. Also updates the view when this is done.

This should solve #5 

(Changes in package-lock are because of `npm audit fix`)

### CODE REVIEW

**Comment by artisticat1** [comment]
Hi, thank you for the PR!

File: main.ts [hunk 1]
```diff
@@ -44,7 +58,7 @@ export default class NLSyntaxHighlightPlugin extends Plugin {
 		const colors = [settings.adjectiveColor, settings.nounColor, settings.adverbColor, settings.verbColor, settings.conjunctionColor];
 
 		for (let i = 0; i < partsOfSpeech.length; i++) {
-			if (enabled[i]) {
```
Comment (artisticat1) [inline main.ts:47]: Rather than simply disabling the styling, can we unload the CodeMirror extension entirely?

You can do this by removing the extension from `NLSyntaxHighlightPlugin.extensions` and calling `app.workspace.updateOptions()`.

See https://github.com/artisticat1/nl-syntax-highlighting/blob/4bee4d127d14e9a9395d867f39424d43ac6a630e/main.ts#L92-L97 for reference.

File: settings.ts [hunk 2]
```diff
@@ -46,7 +48,16 @@ export class NLSyntaxHighlightSettingTab extends PluginSettingTab {
 
 		containerEl.empty();
 
-
+		new Setting(containerEl)
+		.setName("Active")
```
Comment (artisticat1) [inline settings.ts:52]: Can we change this to "Enabled" instead? I feel it's more widely used and understood than "Active".

Comment (Nick-de-Bruin) [reply settings.ts]: Agreed, changed in newest commit

**Comment by artisticat1** [comment]
Thanks! Just a couple more minor points and then I think we're good to merge.

File: main.ts [hunk 3]
```diff
@@ -36,6 +38,28 @@ export default class NLSyntaxHighlightPlugin extends Plugin {
 		await this.saveData(this.settings);
 	}
 
+	async addCommands() {
```
Comment (artisticat1) [inline main.ts:41]: I don't think `addCommands()` itself needs to be async (since `this.addCommand` is synchronous) - it's just the callback function that does

File: main.ts [hunk 4]
```diff
@@ -36,6 +38,28 @@ export default class NLSyntaxHighlightPlugin extends Plugin {
...
+		this.addCommand({
+			id: "toggle-enabled",
+			name: "Toggle natural language highlighting",
```
Comment (artisticat1) [inline main.ts:44]: Obsidian automatically prepends the plugin's name in front of commands, so I suggest we shorten this to just "Toggle on/off"

**Comment by Nick-de-Bruin** [comment]
Should be all done, let me know if there's anything else you think should change!

File: main.ts [hunk 5]
```diff
@@ -17,11 +17,13 @@ export default class NLSyntaxHighlightPlugin extends Plugin {
...
 
-		this.extensions = [NLSyntaxHighlightViewPlugin.extension];
+		this.extensions = this.settings.enabled ? [NLSyntaxHighlightViewPlugin.extension] : [];
 		this.registerEditorExtension(this.extensions);
-
+		
 		this.styleEl = document.head.createEl("style");
 		this.reloadStyle();
+
+		await this.addCommands();
```
Comment (artisticat1) [inline main.ts:26]: Since `addCommands()` is no longer async, we don't need an `await` here anymore :)

Comment (Nick-de-Bruin) [reply main.ts]: Haha that's a bit embarrassing. Fixed

**Comment by artisticat1** [comment]
Looks good! Thanks for your work.
